### ISSUE TITLE
Velocities in spherical coordinates

### ISSUE THREAD

**Issue by al-jshen** [astropy/astropy#11402]
Hi, I'm converting some positions and velocities from ICRS to Galactocentric, and then turning that into a spherical representation. The velocities are given as angular velocities, which is a little strange since I would have expected them in km/s. 

```python
c = coord.SkyCoord(
        ra = ra * u.deg, 
        dec = dec * u.deg, 
        pm_ra_cosdec = pmracosdec * u.mas/u.yr, 
        pm_dec = pmdec * u.mas / u.yr, 
        radial_velocity = vrad * u.km / u.s, 
        distance = dist * u.kpc,
        frame = 'icrs',
    )
cg = c.transform_to(coord.Galactocentric())
cgs = cg.represent_as('spherical')
diffs = cgs.differentials['s']
diffs
<SphericalDifferential (d_lon, d_lat, d_distance) in (mas / yr, mas / yr, kpc mas / (rad yr))
    (-0.17801241, -0.06221044, -12.62912254)>
```
**How can I get these in km/s?** For `d_distance` I can do 
```python
cgs.differentials['s'].d_distance.to(u.km/u.s)
```
but for `d_lon` and `d_lat` I can't do the conversion, even when I multiply by the distance:
```python
(cgs.differentials['s'].d_lon * dist * u.kpc).to(u.km/u.s)
``` 
gives the following error:
```python
UnitConversionError: 'kpc mas / yr' and 'km / s' (speed) are not convertible
```
Thanks.

**Comment by ayshih** [comment]
You need to specify the [`dimensionless_angles()` equivalency](https://docs.astropy.org/en/stable/units/equivalencies.html#angles-as-dimensionless-units) to convert `mas` in the way that you want:
```python
(cgs.differentials['s'].d_lon * dist * u.kpc).to(u.km/u.s, equivalencies=u.dimensionless_angles())
```

**Comment by adrn** [comment]
Yes, for an immediate solution, I use what @ayshih suggests. But it has been a longstanding issue for me that non-cartesian representations don't provide access to velocities in velocity units when possible (similar issues for `PhysicsSpherical` and `Cylindrical` representations). If you have any energy to think about that (specifically what the API should look like for that), please let us know your thoughts! (see also #7655 and cc @mhvk)

**Comment by al-jshen** [comment]
@ayshih @adrn Thanks for the help. Follow-up question: is there anything like this (https://docs.astropy.org/en/stable/coordinates/galactocentric.html) explaining the math for the velocity coordinate transform? Thanks.

**Comment by mhvk** [comment]
About a solution for this: we already have a `SphericalCosLatDifferential` to deal with motion in longitude that includes `cos(lat)` to make it arcsec/yr on the sky. I think it should not be too difficult to make a new differential class, say `SphericalRCosLatDifferential` (better name would be desirable!) that for lat and lon also includes the radial term and thus gives `km/s`. One would then also need to adjust the scale factor calculation, at https://github.com/astropy/astropy/blob/7811614f8b9b08eb3f7bcb1705ad53b186ca8187/astropy/coordinates/representation.py#L1856-L1862

**Comment by adrn** [comment]
Coming back to this because of velocity conversion headaches the students are facing at our Galactic Dynamics summer school — we should really just implement your proposed solution @mhvk (and same for `PhysicsSpherical`, `Cylindrical`, etc.)!

**Comment by mhvk** [comment]
Sounds good! Mostly just needs agreement on the name! Also for the attributes: `v_ra`, `v_dec`, `radial_velocity`, I guess?

**Comment by adrn** [comment]
It could also be nice to have a shorthand, like `c.spherical.velocity`, that has `[v_ra, c_dec, radial_velocity]` as a single quantity, like an analog of `c.cartesian.xyz` - something to think about / discuss on Monday or at the next coordinates call?

**Comment by adrn** [comment]
The specific issue raised was addressed, and the feature request is actually a duplicate of #7655, so closing this!
